.circleci/soltest.sh

```diff
@@ -54,7 +54,13 @@ get_logfile_basename() {
 }
 
 BOOST_TEST_ARGS="--color_output=no --show_progress=yes --logger=JUNIT,error,test_results/`get_logfile_basename`.xml"
-SOLTEST_ARGS="--evm-version=$EVM --evmonepath /usr/lib/libevmone.so $flags"
+if [ "$(uname)" == "Darwin" ]
+then
+  export DYLD_LIBRARY_PATH="/Users/distiller/project/z3/bin"
```

Comment:
i'm not sure what line you mean, but `DYLD_LIBRARY_PATH` is only related to the `z3` library.

Reply:
Are you saying z3 is dynamically linked to solidity?

Reply:
Yes, at least to `soltest`.

Reply:
hmm.. but good point.. i didnt checked the buildsystem.. i was just pointing to the `z3` so that cmake finds it.. looks like that it is not automatically using the static library.. should the static library be used?

Reply:
i guess it may simplify some things.. e.g. `DYLD_LIBRARY_PATH` is not needed.. but currently i got rid of it by just copying the dynamic lib to `/usr/local/lib`..

Reply:
@axic @leonardoalt basically `FindZ3.cmake` is responsible to find the `z3` library with the help of `find_library`.. thats why I set the environment-variable `CMAKE_LIBRARY_PATH` to the directory where the dynamic and static libraries are present.. cmake uses the variable `CMAKE_FIND_LIBRARY_SUFFIXES` to find the correct library, that is by default set to `.so;.a` on unix and `.dylib;.so;.a` on macOS.. I think because a dynamic library is found, dynamic linking is used. I guess if we would change `CMAKE_FIND_LIBRARY_SUFFIXES` to the appropriate static library suffix, the build will then do static linking..  I will try to solve this here by just deleting the `*.dylib` files, then `find_library` should only be able to find the static library, so finally everything should be linked statically..